state/account.go

```diff
@@ -166,6 +171,17 @@ func (st *Account) SetPendingNonce(nonce uint64) error {
 	return nil
 }
 
+// ConvertFreshAccountToZeroNonceType converts a fresh legacy account to zero-nonce account
+func (st *Account) ConvertFreshAccountToZeroNonceType(nonce uint64) bool {
```

Comment:
could this conversion be moved to SetPendingNonce as:
```
    switch st.accountType {
    case 1: ...
    case 0:
        if st.accountType == 0 && st.nonce == 0 && nonce == 1 {
            st.accountType = 1
        } else {
            ....
        }
    default:
        ...
    }
    st.nonce++
```

 The hard fork feature and nonce has been checked in `checkNonceContinuity`, thus we can delete this function as well as the place calling it

Reply:
`SetPendingNonce ` does not have a `ctx` input to tell hard-fork or not, while this conversion is only done after hard-fork. So it's hard to do inside it.